index.jsx

```diff
@@ -1,33 +1,54 @@
-import React from 'react';
-import ReactDOM from 'react-dom/client';
-import { BrowserRouter, Routes, Route, Link } from "react-router-dom"
-import Home from "./pages/Home"
-import About from "./pages/About"
-import Vans from "./pages/Vans"
-import VanDetail from "./pages/VanDetail"
+import React from "react";
+import ReactDOM from "react-dom/client";
+import { RouterProvider, createBrowserRouter, createRoutesFromElements, Route, Link } from "react-router-dom";
+import Home from "./pages/Home";
+import About from "./pages/About";
+import Vans, { loader as vansLoader } from "./pages/Vans/Vans";
+import VanDetail, { loader as vanDetailLoader } from "./pages/Vans/VanDetail";
+import Dashboard from "./pages/Host/Dashboard";
+import Income from "./pages/Host/Income";
+import Reviews from "./pages/Host/Reviews";
+import HostVans, { loader as hostVansLoader } from "./pages/Host/HostVans";
+import HostVanDetail, { loader as hostVanDetailLoader } from "./pages/Host/HostVanDetail";
+import HostVanInfo from "./pages/Host/HostVanInfo";
+import HostVanPricing from "./pages/Host/HostVanPricing";
+import HostVanPhotos from "./pages/Host/HostVanPhotos";
+import NotFound from "./pages/NotFound";
+import Login, { loader as loginLoader, action as loginAction } from "./pages/Login";
+import Layout from "./components/Layout";
+import HostLayout from "./components/HostLayout";
+import Error from "./components/Error";
+import { requireAuth } from "./utils";
 
-import "./server"
+import "./server";
 
-function App() {
-  return (
-    <BrowserRouter>
-      <header>
-        <Link className="site-logo" to="/">#VanLife</Link>
-        <nav>
-          <Link to="/about">About</Link>
-          <Link to="/vans">Vans</Link>
-        </nav>
-      </header>
-      <Routes>
-        <Route path="/" element={<Home />} />
-        <Route path="/about" element={<About />} />
-        <Route path="/vans" element={<Vans />} />
-        <Route path="/vans/:id" element={<VanDetail />} />
-      </Routes>
-    </BrowserRouter>
+const router = createBrowserRouter(
+  createRoutesFromElements(
+    <Route path="/" element={<Layout />}>
+      <Route index element={<Home />} />
+      <Route path="about" element={<About />} />
+      <Route path="login" element={<Login />} loader={loginLoader} action={loginAction} />
+      <Route path="vans" element={<Vans />} errorElement={<Error />} loader={vansLoader} />
+      <Route path="vans/:id" element={<VanDetail />} loader={vanDetailLoader} />
```

Comment:
Please add an error Element on the route so that if there is an error in the loader, you can deal with it appropriately.